hbase-client/src/main/java/org/apache/hadoop/hbase/replication/ReplicationQueuesZKImpl.java

```diff
@@ -132,17 +132,45 @@ public void addLog(String queueId, String filename) throws ReplicationException
   }
 
   @Override
-  public void removeLog(String queueId, String filename) {
+  public void removeLog(String queueId, String filename)
+      throws ReplicationSourceWithoutPeerException {
     try {
-      String znode = ZKUtil.joinZNode(this.myQueuesZnode, queueId);
-      znode = ZKUtil.joinZNode(znode, filename);
-      ZKUtil.deleteNode(this.zookeeper, znode);
-    } catch (KeeperException e) {
-      this.abortable.abort("Failed to remove wal from queue (queueId=" + queueId + ", filename="
-          + filename + ")", e);
+      try {
+        String znode = ZKUtil.joinZNode(this.myQueuesZnode, queueId);
+        znode = ZKUtil.joinZNode(znode, filename);
+        ZKUtil.deleteNode(this.zookeeper, znode);
+      } catch (KeeperException e) {
```

Comment:
We will swallow all the KeeperException which are **not** of` NoNodeException` type. In the internal catch block we can just catch `KeeperException.NoNodeException` and in the else block of `if (!doesPeerExist(queueId)) ` condition we need to rethrow exception so that outer catch block will handle it and abort RS.